controllers/cluster_controller.go

```diff
@@ -280,130 +258,44 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 			}
 
 			if tarantool.IsJoined(pod) {
-				continue
+				reqLogger.Info("Already joined", "Pod.Name", pod.Name)
+				isJoined = true
+				break
 			}
 
-			if err := topologyClient.Join(pod); err != nil {
-				if topology.IsAlreadyJoined(err) {
-					tarantool.MarkJoined(pod)
-					if err := r.Update(context.TODO(), pod); err != nil {
-						return ctrl.Result{RequeueAfter: time.Duration(5 * time.Second)}, err
-					}
-					reqLogger.Info("Already joined", "Pod.Name", pod.Name)
-					continue
-				}
-
-				if topology.IsTopologyDown(err) {
-					reqLogger.Info("Topology is down", "Pod.Name", pod.Name)
-					continue
-				}
-
-				reqLogger.Error(err, "Join error")
-				return ctrl.Result{RequeueAfter: time.Duration(5 * time.Second)}, nil
+			_, exists := replicas[podName]
+			if !exists {
+				reqLogger.Info("Found new node in replicaset, add to hashmap " + podName)
```

Comment:
No need to write about internal structures in logs.

```suggestion
				reqLogger.Info("Found new node in replicaset: " + podName)
```